artifacttemplates/http%3A%2F%2Fopentosca.org%2Fartifacttemplates/DockerEngine_DockerEngine-Interface-w1/source/src/main/java/org/opentosca/artifacttemplates/dockerengine/DockerEngineInterfaceDockerEngineEndpoint.java

```diff
@@ -231,7 +218,25 @@ public void onNext(final BuildResponseItem item) {
 
                     final String[] portMapKV = portMapping.split(",");
                     if (portMapKV.length > 0 && Arrays.stream(portMapKV).noneMatch(String::isEmpty)) {
-                        final ExposedPort tempPort = ExposedPort.tcp(Integer.parseInt(portMapKV[0]));
+                        // exposed port has the pattern <port>/<protocol>, e.g., 9999/udp
+                        String[] portSplit = portMapKV[0].split("/");
+                        String port;
+                        String protocol;
+                        if (portSplit.length == 2) {
+                            port = portSplit[0];
+                            protocol = portSplit[1];
+                        } else {
+                            port = portMapKV[0];
+                            protocol = "tcp";
+                        }
+
+                        ExposedPort tempPort;
```

Comment:
could I map multiple ports on multiple protocols ? I know the nodetype doesn't support that currently, but maybe we need that in the future?

Reply:
Yes, and yes, Miles added this support, so the Node Type supportsthat.

Reply:
Correct me but I did not adjust the Node Type. It is not required.
You can simply pass e.g. 8080/udp as ContainerPort which is simply forwarded by the management bus. 
The only new logic is the parsing of the protocol at the docker engine IA.

Reply:
And I did not adjust any handling of exposing multiple ports. 
My changes should not have any conflicts with existing logic.